### DISCUSSION TITLE
EasyUtilities and its scope

### DISCUSSION THREAD

**Discussion by rozyczko** [easyscience/.github#2] [Ideas]
It would probably make sense to create `EasyUtlities` package with common methods, classes and scripts.
The scope would be different to that in `EasyCore`, which ideally should be only fitting, parameters, base classes etc.

This discussion is to find out which functionality can be refactored to be part of such Utilities package.

Initial ideas:

- SciCat connectors
- Common file format readers/writers

**Comment by rozyczko** [comment]
Pypi already has `EasyUtils` so we should spell it out as `Utilities`

**Comment by AndrewSazonov** [comment]
The scope of `EasyCore` doesn't seem to be precisely defined anywhere :( I like the idea of sharing responsibility between modules, but I'm not sure that the growing number of such modules as independent python packages is a good thing. Maybe we should have one python package called `easyscience` that contains submodules `core`, `utilities`, `fitting`, etc.?

**Comment by rozyczko** [comment]
I agree - we don't need to create new modules for the sake of just having them. I guess, at this point, having scicat connectors and file format parsers is not enough to justify `easyUtilities`.

The current description of easyCore on its pypi site is: 
```
easyCore is the foundation of the easyScience universe, providing the building blocks for libraries and applications which aim to make scientific data simulation and analysis easier.
```

In essence both features mentioned can be considered "building blocks", so maybe we should just tack them onto the existing `easyCore` structure.

**Comment by AndrewSazonov** [comment]
The current description is very general and could include anything, as it does now. EasyCore already has `utils`, so indeed scicat connectors and file format parsers can be implemented there, at least for now.

**Comment by rozyczko** [comment]
Moved to `EasyScience.Utilities` as discussed [here](https://github.com/EasyScience/EasyCore/discussions/217)
